dapr/clients/grpc/client.py

```diff
@@ -318,42 +341,110 @@ def save_state(
             key: str,
             value: Union[bytes, str],
             etag: Optional[str] = None,
-            state_options: Optional[StateOptions] = None,
+            options: Optional[StateOptions] = None,
             metadata: Optional[MetadataTuple] = ()) -> DaprResponse:
         """Saves key-value pairs to a statestore
+
+        This saves a value to the statestore with a given key and state store name.
+        Options for request can be passed with the options field and custom
+        metadata can be passed with metadata field.
+
         The example saves states to a statestore:
             from dapr import DaprClient
             with DaprClient() as d:
                 resp = d.save_state(
                     store_name='state_store'
                     states=[{'key': 'key1', 'value': 'value1'}],
+                    etag='etag',
+                    metadata=(
+                        ('header1', 'value1')
+                    ),
                 )
+
         Args:
             store_name (str): the state store name to save to
-            states (List[dict]): the key-value pairs to be saved
+            key (str): the key to be saved
+            value (bytes or str): the value to be saved
+            etag (str, optional): custom etag to save with
```

Comment:
etag is not a custom etag. etag must be given by backend statestore and used for save and delete. here it is given in the response payload.